src/assets/scripts/client/math/vector.js

```diff
@@ -81,16 +81,46 @@ export const vnorm = (v, length) => {
 };
 
 /**
- * Create a 2D vector
- * Pass a heading (rad) and this will return the corresponding unit vector
+ * Get a 2D unit vector corresponding to a heading in radians
+ *
+ * @param  heading_angle    float    the heading angle, in radians
+ * @return array    an array of the x and y components of the vector
  */
-export const vectorize_2d = (direction) => {
+export const vectorize_2d_from_radians = (heading_angle) => {
+    // keep the angle in the [0, 2*PI[ range
+    while (heading_angle < 0) {
+        heading_angle += 2 * Math.PI;
```

Comment:
I'd encourage you to set `2 * Math.PI` to a variable so you only need to do the math once, and can then reference it every where else